### ISSUE TITLE
WebSocket Endpoint configuration

### ISSUE THREAD

**Issue by artjomg** [ralscha/playground#1]
Hi and Thanks for your examples,

I used the WebSocket Server/Client example but had to adjust the file
https://github.com/ralscha/playground/blob/master/springwebsocket/src/main/java/ch/rasc/springwebsocket/Application.java

According to the hints mentioned here
http://stackoverflow.com/questions/30413380/websocketstompclient-wont-connect-to-sockjs-endpoint

like this:

	@Override
	public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
		RequestUpgradeStrategy upgradeStrategy = new TomcatRequestUpgradeStrategy();
		registry.addHandler(simpleHandler(), "/endpoint");
		registry.addHandler(registryHandler(), "/registry").setHandshakeHandler(new DefaultHandshakeHandler(upgradeStrategy)).setAllowedOrigins("*").withSockJS().setHeartbeatTime(5000);
		registry.addHandler(chatHandler(), "/chat");
		registry.addHandler(chat4Handler(), "/chat4");
		registry.addHandler(quoteHandler(), "/quotes").setHandshakeHandler(new DefaultHandshakeHandler(upgradeStrategy)).setAllowedOrigins("*");
	}

Even for localhost it looks like I had to adjust the handshake strategy by setting it to all "*" for Tomact 8 embedded in SpringBoot.
ArtG

**Comment by ralscha** [comment]
Hi 
Thanks for the feedback

On my computer it works without allowedOrigins when I run the server and client.

```
registry.addHandler(simpleHandler(), "/endpoint");
registry.addHandler(registryHandler(), "/registry").withSockJS().setHeartbeatTime(5000);
registry.addHandler(chatHandler(), "/chat");
registry.addHandler(chat4Handler(), "/chat4");
registry.addHandler(quoteHandler(), "/quotes");
```

Don't know why you had to add allowedOrigins. IMHO it should not be needed when you run the server and client on the same host.


http://docs.spring.io/spring/docs/current/spring-framework-reference/htmlsingle/#websocket-server-allowed-origins

**Comment by artjomg** [comment]
Hi,

might be a Java 7 vs 8 issue or OS specific difference. I am testing on OS X and executing the code in context of eclipse. Which system and/or Java version are you using?

Thx

**Comment by ralscha** [comment]
I tested it on Windows 8.1 with Java 8

I have access to a mac book and a linux server. I will test it on these computer.

**Comment by ralscha** [comment]
Now I also tested in on Mac OS X with Java 8 (1.8.0_45).
It workes without setAllowedOrigins
I started the server (ch.rasc.springwebsocket.Application) then I started the client (ch.rasc.springwebsocket.client.Client) and they are able to speak together. 

Then I tested it with two computers. I started the server on the mac book (192.168.178.63) and then changed the addresses in Client.java (ws://192.168.178.63:8080/) on my windows computer and it still works without setAllowedOrigins

What the spring framework does is checking the Origin header.  The java client sends the server address in the Origin Header. This way it works without setAllowedOrigins

You find the code that checks the Origin header here: org.springframework.web.util.WebUtils#isValidOrigin

I tested this with Spring 4.1.7. Do you use a different version?

**Comment by artjomg** [comment]
Sorry, missed to mention that I am using  Spring Boot  (v1.2.4.RELEASE) which internally uses 4.1.6 Spring release. And one more difference is that I am using localhost as hostname.
But changing it to a local IP still guides to

```
19:41:55.658 [SimpleAsyncTaskExecutor-2] ERROR o.s.w.s.s.c.DefaultTransportRequest - No more fallback transports after TransportRequest[url=ws://xxx.xxx.xxx.xxx:8080/registry/147/d137893e54ee4fa5822498e829db3685/websocket]
javax.websocket.DeploymentException: Handshake error.
	at org.glassfish.tyrus.client.ClientManager$3$1.run(ClientManager.java:674) ~[tyrus-standalone-client-1.11.jar:na]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_25]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_25]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[na:1.8.0_25]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[na:1.8.0_25]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_25]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_25]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_25]
Caused by: org.glassfish.tyrus.core.HandshakeException: Response code was not 101: 403.
	at org.glassfish.tyrus.client.TyrusClientEngine.processResponse(TyrusClientEngine.java:320) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.tyrus.container.grizzly.client.GrizzlyClientFilter.handleHandshake(GrizzlyClientFilter.java:346) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.tyrus.container.grizzly.client.GrizzlyClientFilter.handleRead(GrizzlyClientFilter.java:315) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565) ~[tyrus-standalone-client-1.11.jar:na]
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545) ~[tyrus-standalone-client-1.11.jar:na]
	... 1 common frames omitted
```

Changing to setAllowedOrigins executes successfully. :-(

**Comment by ralscha** [comment]
Looks like this is the problem. I can reproduce your problem with Spring Boot 1.2.4 (Spring 4.1.6)
It only works when I add ```setAllowedOrigins("*")```

With Spring Boot 1.2.5 (Spring 4.1.7) this problem disappears and setAllowedOrigins is not necessary.

I found the corresponding JIRA issue they fixed in 4.1.7
https://jira.spring.io/browse/SPR-12956

**Comment by artjomg** [comment]
Sorry for the trouble and thanks for your support. 
Cheers from Germany.
